File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
import java.util.Hashtable;
import java.util.HashSet;
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Hashtable<Integer,Integer> numOfOcc = new Hashtable<>();
        HashSet<Integer> uniqOcc = new HashSet<Integer>(); 
        int lenght = arr.length;
        
        for (int i : arr) {
            numOfOcc.put(i, numOfOcc.getOrDefault(i, 0) + 1);
        }

        for(int count :numOfOcc.values()){
            if(!uniqOcc.add(count)){
                return false;
            }                      
        }
        return true;
    }
}

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        freq_dict = {}

        for num in arr:
            freq_dict[num] = freq_dict.get(num,0)+1
            
        
        val = freq_dict.values()
        set_val = set(val)

        return len(val) == len(set_val)